include/one_dimensional_linear_interpolator.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef __ONE_DIMENSIONAL_LINEAR_INTERPOLATOR_H__
+#define __ONE_DIMENSIONAL_LINEAR_INTERPOLATOR_H__
+
+#include "interpolator.hpp"
+#include <CL/sycl.hpp>
+#include <mpi.h>
+#include <neso_particles.hpp>
+#include <vector>
+
+using namespace NESO::Particles;
+using namespace cl;
+namespace NESO {
+
+/**
+ *  Class used to output a vector of y values, given some x values, based
+ * on provided x,y values for input
+ */
+
+class OneDimensionalLinearInterpolator : public Interpolator {
+public:
+  OneDimensionalLinearInterpolator(std::vector<double> x_data,
+                                   std::vector<double> y_data,
+                                   SYCLTargetSharedPtr sycl_target)
+      : Interpolator(x_data, y_data, sycl_target){
+
+        };
+
+  virtual void get_y(std::vector<double> &x_input,
+                     std::vector<double> &y_output) {
+    interpolate(x_input, y_output);
+  }
+
+protected:
```

Comment:
The automatic API documentation relies on there being "docstrings" that describe what the function does and what the parameters are for, e.g. at minimum

```
/**
 * Description of what this method does
 * 
 * @param[in] x_input What is x_input
 * @param[in, out] y_output What is y_output.
 */

```